### PR TITLE
Replace existing structure with a CRA typescript template

### LINKED ISSUE #5: Replace existing structure with a CRA typescript template
## Why

The current template version is already outdated so we decided to remove all files and start a new version using  [CRA typescript template](https://github.com/facebook/create-react-app/tree/master/packages/cra-template-typescript)

more info about creating a custom CRA template can be found on [this document](https://create-react-app.dev/docs/custom-templates)

## Who Benefits?

developers who work on React App

### PR DESCRIPTION
**Pull request by carryall**
## What happened
Resolve #5 

- replace existing files with the [CRA Typescript Template](https://github.com/facebook/create-react-app/tree/master/packages/cra-template-typescript)
- update readme
- use specific package versions
 
## Insight
N/A
 

## Proof Of Work
Test locally by running

`npx create-react-app my-react-app --template file:<path-to-the-template>`

and have the React project created

Screen Shot 2564-04-30 at 15 04 54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,87 +1,20 @@
-<p align="center"><img src="https://cloud.githubusercontent.com/assets/3129129/22811426/bb69dc06-ef0c-11e6-8092-a0bea9060b35.png"/></p>
+# cra-template-nimble
```
Comment (rafayet-monon) [inline README.md:1]: Should we add the [license and about section](https://github.com/nimblehq/rails-templates#license) in this PR or create another task to modify Readme?

File: package.json [hunk 2]
```diff
@@ -1,116 +1,26 @@
...
-    "hmr",
-    "live",
-    "edit",
-    "webpack",
-    "redux",
-    "boilerplate",
-    "starter"
+    "create-react-app",
+    "template",
+    "typescript"
```
Comment (rafayet-monon) [inline package.json:8]: Should we add a "nimble" keyword as well?

File: README.md (hunk 1 above)
Comment (rafayet-monon) [reply README.md:1]: @carryall there's a topmost section before [Get started](https://github.com/nimblehq/rails-templates#get-started) with Nimble logo and project Name. We can add it too.

Comment (carryall) [reply README.md:1]: ✅  added in [fa2a7a0](https://github.com/nimblehq/react-templates/pull/12/commits/fa2a7a0ebd6fad3dac3e3dbd79bd67e4b9ff55b1)

File: package.json (hunk 2 above)
Comment (carryall) [reply package.json]: ✅  added in [7f1c1ce](https://github.com/nimblehq/react-templates/pull/12/commits/7f1c1ce3ea1a3d6ec6996ee3f76b9ac9a70d01a5)

**Review by rafayet-monon — APPROVED** [review]
🚀

**Review by noor-ahmed — APPROVED** [review]

**Pull request merged**